### DISCUSSION TITLE
Can Axum LiveView support third-party templating engines—including run-time ones, like Ramhorns?

### DISCUSSION THREAD

**Discussion by dsaghliani** [davidpdrsn/axum-live-view#155] [Q&A]
Everything seems to be built around the `Html` struct (and `html!` macro). Its documentation is scarce, so I don't know if it's supposed to serve as an interface between HTML code and the crate or as a wholesale templating engine.

There's a hacky way to extract the HTML code into its own file, which may be preferable to many users (due to better editor support), so theoretically, it should be possible to integrate compile-time engines, like Askama, at least with a bit of help from the crate (since `DynamicFragment` seems to play a prominent role, and it's sequestered away in a private module.)

```rs
fn render(&self) -> Html<Self::Message> {
    html! {
        { include_str!("counter.html") }
    }
}
```

But what about run-time engines? Does the crate as it currently is support that, or would the templating modules require an overhaul?

**Comment by davidpdrsn** [comment]
I exists to generate HTML code that can be diff'ed, such that we only send minimal changes cross the socket.

I'm currently not planning on integrating other templating engines.
